backend/utils/external_service/errors.py

```diff
@@ -16,6 +16,10 @@
 class ExternalServiceError(Exception):
     """Base class for exceptions related to external services."""
 
+    def __init__(self, message: str = "", *, extra_data: Any = None) -> None:
+        super().__init__(message)
+        self.extra_data = extra_data
+
```

Comment:
I would create custom exceptions for the `HelsinkiProfileClient` rather than extend the base exception for all clients